test/components/practice.spec.coffee

```diff
@@ -0,0 +1,148 @@
+{expect} = require 'chai'
+_ = require 'underscore'
+React = require 'react/addons'
+Router = require 'react-router'
+Promise = require('es6-promise').Promise
+
+{CourseActions, CourseStore} = require '../../src/flux/course'
+{TaskActions, TaskStore} = require '../../src/flux/task'
+{TaskStepActions, TaskStepStore} = require '../../src/flux/task-step'
+
+{SinglePractice, Tasks} = require '../../src/components'
+{routes} = require '../../src/router'
+
+VALID_MODEL = require '../../api/courses/1/practice.json'
+
+
+div = document.createElement('div')
+
+routerHelper = (route) ->
+  history = new Router.TestLocation([route])
+
+
+  promise = new Promise (resolve, reject) ->
+    Router.run routes, history, (Handler, state)->
+      router = @
+      React.render(<Handler/>, div, ()->
+        component = @
+        result = {div, component, state, router, history}
+        resolve(result)
+      )
+
+tasksHelper = (courseId) ->
+  routerHelper('/courses/' + courseId + '/tasks/')
+
+courseHelper = (model, courseId) ->
+  # Load practice in CourseStore
+  CourseActions.loaded(model, courseId)
+  routerHelper('/courses/' + courseId + '/practice/')
+
+
+
+exerciseTestActions = 
+  fillFreeResponse: (node) ->
+    textarea = node.querySelector('textarea')
+    textarea.value = 'Test Response'
+    React.addons.TestUtils.Simulate.focus(textarea)
+    React.addons.TestUtils.Simulate.keyDown(textarea, {key: 'Enter'})
+    React.addons.TestUtils.Simulate.change(textarea)
+
+    textarea
+
+  clickContinueButton: (node) ->
+    continueButton = node.querySelector('button.btn-primary')
+    React.addons.TestUtils.Simulate.click(continueButton)
+    continueButton = node.querySelector('button.btn-primary')
+
+
+
+describe 'Practice Widget', ->
+  beforeEach ->
+    CourseActions.reset()
+  afterEach ->
+    React.unmountComponentAtNode(div)
+
+  it 'should load the practice button on the course tasks page', (done) ->
+    tests = ({div}) ->
+      buttons = Array.prototype.slice.call(div.querySelectorAll('button.btn-primary'))
+      practiceButton = _.last(buttons)
+      expect(practiceButton.innerText).to.equal('Practice')
```

Comment:
Can you test on class names (`-practice`) rather than the text? Gives Fred more opportunities to style the pages (he'll go through and unprefix the class if he plans on using it)